packages/clerk-js/src/ui/Components.tsx

```diff
@@ -258,7 +276,7 @@ const Components = (props: ComponentsProps) => {
             <VirtualRouter
               preservedParams={PRESERVED_QUERYSTRING_PARAMS}
               onExternalNavigate={() => componentsControls.closeModal('userProfile')}
-              startPath='/user'
+              startPath={'/user' + urlStateParam?.path}
```

Comment:
🔧 Instead of string concatenation I'd suggest using the `buildUrl` helper that ensures proper & safer URL building.